### ISSUE TITLE
CFRunLoop required (?)

### ISSUE THREAD

**Issue by smilingthax** [Syphon/Syphon-Framework#75]
Syphon uses `NSDistributedNotificationCenter` (e.g.) to detect already running servers, which requires a spinning `CFRunLoop` (e.g via `NSRunLoop`). 
Esp. when using a non-GUI Server one might be tempted to skip this.

For Syphon Servers AFAICT only Clients requesting initial discovery of already running servers does not work, when the Server does not use `CFRunLoop` ...
Maybe this requirement could be relaxed in Syphon?

Whereas for Syphon Clients / Discovery, receiving Notifications is at the moment a fundamental requirement.

At least, this should be documented.

**Comment by smilingthax** [comment]
For example, when using the GLib/GTK+ cross-platform framework: `Glib::MainLoop` does not spin CFRunLoop, vs `Gtk::Main` / `Gtk::Application`, which does...  (cf. https://github.com/Syphon/Syphon-Framework/pull/56#issuecomment-1227390622)

**Comment by vade** [comment]
I dont think this will be relaxed w/o a major refactor, and a new messaging method tbh.
